service/service_test.go

```diff
@@ -180,6 +199,12 @@ func TestClose(t *testing.T) {
 			err := svc.Init(ctx, cfg, testBuildTime, testGitCommit, testVersion)
 			So(err, ShouldBeNil)
 
+			// report kafka channels ready to prevent blocking on service start
+			go func(){
+				svc.producer.Channels().Initialised <- struct{}{}
```

Comment:
I would suggest closing the channels instead of sending an empty struct, so that it acts like a `broadcast` , otherwise, if something else is waiting on those channels, only one of the `listeners` would receive it.
In the real dp-kafka code, the initialised channel is closed, and the state channels are closed when a state is reached and re-created when it's left.